x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/UnaryScalarFunction.java

```diff
@@ -47,18 +46,4 @@ public final Expression field() {
     public DataType dataType() {
         return field.dataType();
     }
-
-    @Override
-    public final int hashCode() {
-        return Objects.hash(field);
```

Comment:
The implementation in `Function` also hashes the class name, likely causing fewer collisions.